README.md

```diff
@@ -308,6 +309,22 @@ The backend system would collect all `sensu_check` resources.
   Sensu_check <<||>>
 ```
 
+### Hiera resources
+
+All the types provided by this module can have their resources defined via Hiera. A type such as `sensu_check` would be defined via `sensu::backend::checks`.
+
+The following example adds a check via Hiera:
+
+```yaml
+sensu::backend::checks:
+  check-cpu:
+    ensure: present
+    command: check-cpu.sh -w 75 -c 90
+    interval: 60
+    subscriptions:
+      - linux
```

Comment:
add another common one here to drive home that this work for any type and provider

Reply:
Added some examples based on real world usage and examples from sensu-go docs.